digitalocean/resource_digitalocean_spaces_bucket.go

```diff
@@ -106,6 +110,116 @@ func resourceDigitalOceanBucket() *schema.Resource {
 				Description: "The FQDN of the bucket",
 				Computed:    true,
 			},
+
+			"lifecycle_rule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringLenBetween(0, 255),
+						},
+						"prefix": {
+							Type:     schema.TypeString,
+							Optional: true,
```

Comment:
Specifying a prefix starting with `/` is almost never wanted and leads to a good bit of user error. Though there are some edge cases where it might be wanted. So we should log a a non fatal error here. 

```suggestion
							Optional: true,
							ValidateFunc: func(v interface{}, k string) (ws []string, es []error) {
								if strings.HasPrefix(v.(string), "/") {
									ws = append(ws, "prefix begins with `/`. In most cases, this should be excluded.")
								}
								return
							},
```